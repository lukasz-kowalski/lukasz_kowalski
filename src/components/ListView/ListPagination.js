import React from 'react';
import { Link } from 'react-router-dom';

const ListPagination = ({ isFirstPage, isLastPage, numCurrentPage ,paginationData }) => (
   <nav aria-label="Data grid navigation">
    <ul className="pagination justify-content-end">
      <li className={`page-item ${isFirstPage ? 'disabled' : ''}`}>
        <Link 
          to={{ pathname: `/characters/${numCurrentPage - 1}`, search: `_page=${numCurrentPage - 1}`}}
          className="page-link" 
          tabIndex={isFirstPage ? '-1' : '0'}
        >
          Previous
        </Link>
      </li>

      {paginationData.map(link => link)}

      <li className={`page-item ${isLastPage ? 'disabled' : ''}`}>
        <Link 
          to={{ pathname: `/characters/${numCurrentPage + 1}`, search: `_page=${numCurrentPage + 1}`}}
          className="page-link" 
          tabIndex={isLastPage ? '-1' : '0'}
        >
          Next
        </Link>
      </li>
    </ul>
  </nav>
);


export default ListPagination;
