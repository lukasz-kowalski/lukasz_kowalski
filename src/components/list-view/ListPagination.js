import React from 'react';
import { Link } from 'react-router-dom';

const ListPagination = ({ isFirstPage, isLastPage, numCurrentPage ,paginationData, queryParam }) => (
   <nav aria-label="Data grid navigation">
    <ul className="pagination justify-content-end">
      <li className={`page-item ${isFirstPage ? 'disabled' : ''}`}>
        <Link 
          to={`/characters/${numCurrentPage - 1}`} 
          className="page-link" 
          tabIndex={isFirstPage ? '-1' : '0'}
        >
          Previous
        </Link>
      </li>

      {paginationData.map(link => link)}

      <li className={`page-item ${isLastPage ? 'disabled' : ''}`}>
        <Link 
          to={`/characters/${numCurrentPage + 1}${queryParam}`} 
          className="page-link" 
          tabIndex={isLastPage ? '-1' : '0'}
        >
          Next
        </Link>
      </li>
    </ul>
  </nav>
);

export default ListPagination;
