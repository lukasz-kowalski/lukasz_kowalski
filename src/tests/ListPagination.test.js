import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';
import ListPagination from '../components/ListView/ListPagination';

let wrapper;

const numCurrentPage = 1;
const i = 1;
const paginationData = [
  <li key={'nav-li' + i} className="page-item">
    <NavLink
      to={{ pathname: `/characters/${i}`, search: `_page=${i}`}}
      className={`page-link ${numCurrentPage === i ? 'active' : ''}`}
    >
      {i} <span className="sr-only">{numCurrentPage === i ? '(current)' : ''}</span>
    </NavLink>
  </li>
]

beforeEach(() => {
  wrapper = shallow(
  <ListPagination
    isFirstPage={true}
    isLastPage={false}
    numCurrentPage={numCurrentPage}
    paginationData={paginationData}
  />);
})

test('should successful render ListPagination', () => {
  expect(wrapper).toMatchSnapshot();
})
