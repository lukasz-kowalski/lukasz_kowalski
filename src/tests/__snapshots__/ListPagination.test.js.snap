// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render ListPagination 1`] = `
<nav
  aria-label="Data grid navigation"
>
  <ul
    className="pagination justify-content-end"
  >
    <li
      className="page-item disabled"
    >
      <Link
        className="page-link"
        replace={false}
        tabIndex="-1"
        to={
          Object {
            "pathname": "/characters/0",
            "search": "_page=0",
          }
        }
      >
        Previous
      </Link>
    </li>
    <li
      className="page-item"
      key="nav-li1"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="page-link active"
        to={
          Object {
            "pathname": "/characters/1",
            "search": "_page=1",
          }
        }
      >
        1
         
        <span
          className="sr-only"
        >
          (current)
        </span>
      </NavLink>
    </li>
    <li
      className="page-item "
    >
      <Link
        className="page-link"
        replace={false}
        tabIndex="0"
        to={
          Object {
            "pathname": "/characters/2",
            "search": "_page=2",
          }
        }
      >
        Next
      </Link>
    </li>
  </ul>
</nav>
`;
